scripts/install/installnode.py

```diff
@@ -0,0 +1,512 @@
+                "instance.agentcontroller.client.login": "node",
+                "instance.agentcontroller.client.passwd": "",
+                "instance.agentcontroller.client.port": "4444",
+                },
+                "domain": 'jumpscale',
+                "servicename": "agentcontroller_client",
+                "instance": 'main'
+            }
+
+    def install(self):
+        self.install_packages()
+        self.install_redis()
+        self.install_osis_client()
+        self.install_agentcontroller_client()
+        self.install_jsagent()
+        # self.install_nginx()
+    
+    def install_redis(self):
+        config = self.redis['config']
+        j.clients.redis.configureInstance(
+            "{}__{}".format(self.redis['servicename'], self.redis['instance']),
+            ip=config['ip'],
+            port=config['port'],
+            maxram=config['maxram'],
+            appendonly=False,
+            unixsocket=config['unixsocket']
+            )
+        self.register_service(self.redis)
+        # self.register_service(self.redis2)
+
+    def install_jsagent(self):
+        hpath = j.system.fs.joinPaths(j.dirs.hrdDir, 'system', 'grid.hrd')
+        if not j.system.fs.exists(path=hpath):
+            hrd = j.core.hrd.get(hpath)
+            hrd.set('id', self.config.grid_id)
+            hrd.set('node.id', '0')
+            hrd.set('node.machineguid', j.application.getUniqueMachineId())
+            hrd.set('node.roles', self.config.roles)
+            hrd.save()
+        self.write_hrd(self.jsagent)
+        self.register_service(self.jsagent)
+        # reload system config / whoAmI
+        j.application.loadConfig()
+
+    # def install_nginx(self):
+    #     j.do.execute('apt-get install -y libgd3')
+    #     j.system.fs.createDir(j.system.fs.joinPaths('/var', 'lib', 'nginx'))
+    #     logPath = j.system.fs.joinPaths('/var', 'log', 'nginx')
+    #     if not j.system.fs.exists(logPath):
+    #         j.system.fs.createDir(logPath)
+        # self.register_service(self.nginx)
+
+    def install_osis_client(self):
+        self.write_hrd(self.osis_client_main)
+        self.write_hrd(self.osis_client_jsagent)
+    
+    def install_agentcontroller_client(self):
+        self.write_hrd(self.agent_controller_client)
+
+    def register_service(self, config):
+        j.system.platform.ubuntu.serviceInstall("{}__{}".format(config['servicename'], config['instance']), config['daemonpath'], config['args'], pwd=config['pwd'])
+        j.system.platform.ubuntu.startService("{}__{}".format(config['servicename'], config['instance']))
+
+    def restart_service(self, servicename):
+        j.system.platform.ubuntu.startService(servicename)
+
+    def write_hrd(self, data):
+        hrdname = "{domain}__{service}__{instance}.hrd".format(domain=data['domain'], service=data['servicename'], instance=data['instance'])
+        hpath = j.system.fs.joinPaths(j.dirs.hrdDir, 'apps', hrdname)
```

Comment:
this name is not correct
it should be
`{domain}__{service}__{instance}/service.hrd`